kotlin/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.kt

```diff
@@ -0,0 +1,354 @@
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a surface for elevation data
+    val localElevationImageService = getString(R.string.elevation_service)
+    val surface = Surface().apply {
+      elevationSources.add(ArcGISTiledElevationSource(localElevationImageService))
+    }
+
+    // create a layer of buildings
+    val buildingsLayer = ArcGISSceneLayer(getString(R.string.buildings_layer))
+
+    // create a scene and add imagery basemap, elevation surface, and buildings layer to it
+    val scene = ArcGISScene().apply {
+      basemap = Basemap.createImagery()
+      baseSurface = surface
+      operationalLayers.add(buildingsLayer)
+    }
+
+    val initLocation = Point(-4.50, 48.4, 1000.0)
+    Viewshed.setFrustumOutlineColor(Color.BLUE)
+
+    // create viewshed from the initial location
+    viewShed = LocationViewshed(
+      initLocation,
+      initHeading.toDouble(),
+      initPitch.toDouble(),
+      initHorizontalAngle.toDouble(),
+      initVerticalAngle.toDouble(),
+      initMinDistance.toDouble(),
+      initMaxDistance.toDouble()
+    ).apply {
+      setFrustumOutlineVisible(true)
+    }
+
+    // add a camera and set it to orbit the location point of the frustum
+    val camera = Camera(initLocation, 20000000.0, 0.0, 55.0, 0.0)
+    val orbitCamera = OrbitLocationCameraController(initLocation, 5000.0)
+    sceneView.apply {
+      this.scene = scene
+      this.cameraController = orbitCamera
+      this.setViewpointCamera(camera)
+    }
+
+    // create an analysis overlay to add the viewshed to the scene view
+    val analysisOverlay = AnalysisOverlay()
+    analysisOverlay.analyses.add(viewShed)
+    sceneView.analysisOverlays.add(analysisOverlay)
+
+    // initialize the UI controls
+    handleUiElements()
+  }
+
+  /**
+   * Handles double touch drag for movement of viewshed location point and listeners for
+   * changes in seek bar progress.
+   */
+  private fun handleUiElements() {
+
+    sceneView.setOnTouchListener(object : DefaultSceneViewOnTouchListener(sceneView) {
+
+      // double tap and hold second tap to drag viewshed to a new location
+      override fun onDoubleTouchDrag(motionEvent: MotionEvent): Boolean {
+        // convert from screen point to location point
+        val screenPoint = android.graphics.Point(
+          motionEvent.x.roundToInt(),
+          motionEvent.y.roundToInt()
+        )
+        val locationPointFuture = sceneView.screenToLocationAsync(screenPoint)
+        locationPointFuture.addDoneListener {
+          try {
+            val locationPoint = locationPointFuture.get()
+
+            // add 50 meters to location point and set to viewshed
+            viewShed.location = Point(locationPoint.x, locationPoint.y, locationPoint.z + 50)
+          } catch (e: InterruptedException) {
+            val error = "Error converting screen point to location point: " + e.message
```

Comment:
I likes the .also pattern you used on other samples...could be good here again!

Reply:
I changed it up a bit and added a method to log and toast, like the contributor's guide says, and used let{} so i could easily handle null strings.